Library/Homebrew/test/support/helper/spec/shared_examples/formulae_exist.rb

```diff
@@ -2,7 +2,7 @@
 
 RSpec.shared_examples "formulae exist" do |array|
   array.each do |f|
-    it "#{f} formula exists" do
+    it "#{f} formula exists", :needs_homebrew_core do
       core_tap = Pathname("#{HOMEBREW_LIBRARY_PATH}/../Taps/homebrew/homebrew-core")
```

Comment:
Feels like this could look in the JSON instead for this key.

Reply:
Technically yes but I don't know a simple way to set that up in the test environment.

Reply:
Write a mock API JSON to the cache.

Reply:
I'm pretty sure that it would require writing a JSON file signed with a private key to the cache along with mocking the public key so that we can check that the JSON file is correct when loading it for the first time. The same thing would need to happen with the formula tap migrations file as well or that would need to be mocked. We'd want to decide before that where we should put the public and private keys in the repo and if we want to sign things on the fly or have fixtures for the signed JSON files. It's not as simple as just writing one file to the cache.

Reply:
> That's only necessary if you want to avoid any Ruby mocking. You should be able to mock/populate the cache of higher-level methods instead e.g. `Homebrew::API::Formula.all_formulae`, `tap_migrations`.

Yeah, mocking those method calls is simple enough to do. We already do that in a few places in tests. We'd just have create a larger API JSON test fixture to read from.

I think I got confused by the `Write a mock API JSON to the cache.` comment which implied to me that we wanted to actually write a file to the cache directory which would require more of the stuff I mentioned before.

Either way, discussion about how to best add tests for the API path is better done here: https://github.com/Homebrew/brew/issues/16895

I'm thinking that some of these problems could be solved by test helpers and convention.